DOL.WHD.Section14c.Test/Helpers/NLoggerTests.cs

```diff
@@ -20,37 +22,43 @@ public class NLoggerTests
-            httpRequestMessage.Content = new StringContent("", Encoding.UTF8, "application/x-www-form-urlencoded");
-
-            var httpActionContext = new HttpActionContext
+            try
             {
-                ControllerContext = new HttpControllerContext
-                {
-                    Request = httpRequestMessage,
-                    ControllerDescriptor = new HttpControllerDescriptor()
-                },
-                Response = new HttpResponseMessage
-                {
-                    StatusCode = HttpStatusCode.Conflict,
-                    Content = new StringContent("test")
-                }
+                var correlationId = Guid.NewGuid().ToString();
+                var httpRequestMessage = new HttpRequestMessage(HttpMethod.Post, "mywebapi/test");
+                httpRequestMessage.Content = new StringContent("", Encoding.UTF8, "application/x-www-form-urlencoded");
 
-            };
-            var mockLogger = new LoggerMock();
-            GlobalConfiguration.Configuration.Services.Replace(typeof(ITraceWriter), mockLogger);
-            var trace = GlobalConfiguration.Configuration.Services.GetTraceWriter();
-            httpActionContext.Request.Properties[Constants.CorrelationId] = correlationId;
-            trace.Info(httpActionContext.Request,
-                "Category",
-                "JSON", 
-                "ActionArguments");
-
-
-            Assert.AreEqual("Category", mockLogger.LogToOperationsCalledWithMessage);
+                var httpActionContext = new HttpActionContext
+                {
+                    ControllerContext = new HttpControllerContext
+                    {
+                        Request = httpRequestMessage,
+                        ControllerDescriptor = new HttpControllerDescriptor()
+                    },
+                    Response = new HttpResponseMessage
+                    {
+                        StatusCode = HttpStatusCode.Conflict,
+                        Content = new StringContent("test")
+                    }
+                };          
+                
+                var loggerMock = new Mock<ILogger>();
 
+                // Create Mock for NLog
+                // NLog is unit tested in NLog itself. No need to retest it.
+                // If there is no exception, test consider passed
+                GlobalConfiguration.Configuration.Services.Replace(typeof(ITraceWriter), new NLogger(loggerMock.Object));
+                var trace = GlobalConfiguration.Configuration.Services.GetTraceWriter();
+                httpActionContext.Request.Properties[Constants.CorrelationId] = correlationId;
+                trace.Info(httpActionContext.Request,
+                    "Category",
+                    "JSON",
+                    "ActionArguments");                
+            }
+            catch(Exception e)
+            {
+                Assert.Fail();
+            }            
```

Comment:
This ends up with the same `ITraceWriter` testing problem as elsewhere, with no good workaround.  That said, this is a nice change because it does incorporate the `NLogger` class now!